core/trie/node.go

```diff
@@ -109,9 +114,19 @@ func (n *Node) UnmarshalBinary(data []byte) error {
 		var pathP **bitset.BitSet
 		switch head {
 		case 'l':
-			pathP = &(n.left)
+			if !slot.leftSlot {
```

Comment:
you dont need to do any bookkeeping. checking if `n.left` or `n.right`, respectively, is not `nil` should be enough

Reply:
Ohh, your solution is  more clean. Let me update it!